### PR TITLE
3.5_GET_/api Completed

### PR DESCRIPTION
**Pull request by Tucker-Ajai**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -37,4 +37,17 @@ describe("Get /api/categories", () => {
...
   });
+
+  describe("3.5 GET /api", () => {
+    test("", () => {
+      return request(app)
+        .get("/api")
+        .expect(200)
+        .then((response) => {
+          console.log(Object.keys(response.body))
+          expect(Object.keys(response.body).includes('GET /api')).toBe(true)
```
Comment (Beth-McCormack) [inline __tests__/app.test.js]: This test will not be robust enough when you add to the endpoints. If you require the endpoints.json into this file, think about how you can compare this to the response to ensure they look the same.

File: Models/api.model.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const endpoint = require("../endpoints.json");
+
+exports.fetchApi = () => {
```
Comment (Beth-McCormack) [inline Models/api.model.js]: As you're not interacting with the database you dont need anything in the model for this ticket.

File: Controllers/api.controller.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const express = require("express");
+const { fetchApi } = require("../Models/api.model");
+
+const app = express();
+
+exports.getApi = (request, response, next) => {
+  
+    response.status(200).send(fetchApi());
```
Comment (Beth-McCormack) [inline Controllers/api.controller.js]: .send() should send the final response to the client. Sending a function here means theres no guarantee what will be sent back, and is very unreliable if the fetchApi function changes at all in the future.

**Review by Beth-McCormack — CHANGES_REQUESTED** [review]
Please see comments and submit another PR when done 🙂

File: Models/api.model.js (hunk 2 above)
Comment (Tucker-Ajai) [reply Models/api.model.js]: I have removed everything from the model now

File: Controllers/api.controller.js (hunk 3 above)
Comment (Tucker-Ajai) [reply Controllers/api.controller.js]: I have  now required in the JSON file straight to the controller and now sending that as the response

File: __tests__/app.test.js (hunk 1 above)
Comment (Tucker-Ajai) [reply __tests__/app.test.js]: It didn't occur to me to require the JSON into the test file and so I was checking for the 1 thing that will always be consistent. I will now change it to check against the file required into test vs the file required into the controller

**Review by Beth-McCormack — APPROVED** [review]

**Pull request merged**